### PR TITLE
add example for multi-arch binary

### PR DESCRIPTION
**Pull request by juliexxia**
Final step for https://github.com/bazelbuild/bazel/issues/5575

Note - this logic isn't actually released in bazel yet but I think it's fine to document it prematurely.

### CODE REVIEW

**Comment by juliexxia** [comment]
I guess checks won't pass until the logic is in bazel...

File: rules/starlark_configurations/multi_arch_binary/defs.bzl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    # dependencies with providers.
+    x86_dep = tools['x86-platform']
+    armeabi_v7a_dep = tools['armeabi-v7a-platform']
+    return []
+
+foo_binary = rule(
+  implementation = _rule_impl,
+  attrs = {
+    "tool": attr.label(cfg = fat_transition),
+    '_whitelist_function_transition': attr.label(
```
Comment (gregestren) [inline rules/starlark_configurations/multi_arch_binary/defs.bzl:36]: Let's comment why this is here, and in every other example that uses this attribute. I know users have been confused by it before.

Do we have a good canonical link to reference that explains this?

File: rules/starlark_configurations/multi_arch_binary/defs.bzl [hunk 2]
```diff
@@ -0,0 +1,45 @@
+def _transition_impl(settings, attr):
+    _ignore = [settings, attr]
+
+    # Return a dict of dicts. The values are the updates to the configuration.
+    # The keys are arbitrary helpful strings that can be used to access the split
+    # targets in the rule context
+    #
+    # Returning a dict of dicts creates a "split transition", which transitions
```
Comment (gregestren) [inline rules/starlark_configurations/multi_arch_binary/defs.bzl:8]: Link to the split transition docs section too?

File: rules/starlark_configurations/multi_arch_binary/defs.bzl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    # the dep its attached to to more than one configuration creating multiple
```
Comment (gregestren) [inline rules/starlark_configurations/multi_arch_binary/defs.bzl:9]: typo nit: it's

File: rules/starlark_configurations/multi_arch_binary/defs.bzl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    outputs = ["//command_line_option:cpu"]
+)
+
+def _rule_impl(ctx):
+    # Access the split dependencies via `ctx.split_attr.<split-attr-name>`
+    tools = ctx.split_attr.tool
+    # The values of `x86_dep` and `armeabi-v7a_dep` here are regular
+    # dependencies with providers.
+    x86_dep = tools['x86-platform']
+    armeabi_v7a_dep = tools['armeabi-v7a-platform']
```
Comment (gregestren) [inline rules/starlark_configurations/multi_arch_binary/defs.bzl:29]: I have an idea that makes the example a bit more elaborate, but could really drive the point home by showing explicitly in the output what's going on.

What about having `_impl` below return a provider with a string based on its CPU, like `SimpleProvider(my_info= "My CPU is X")`. Then have `print()` statements in `_rule_impl` that say `"tools['x86-platform'] provider says: <whatever that instance of the provider says"`.

If it's unclear what I'm saying, let me know and I'll share some more specific code snippets to demonstrate.

**Comment by gregestren** [comment]
Oh, and if `bazelbuild/examples` automatically tries to build whatever gets checked in, will we have a problem if it's not released yet?

**Comment by gregestren** [comment]
Check out https://github.com/bazelbuild/bazel/issues/11077. This might actually get released soon.

**Comment by juliexxia** [comment]
I temporarily disabled it with a TODO back to this issue to re-enable when the release happens

**Comment by juliexxia** [comment]
Added cpu printouts!

**Pull request merged**